uni/lib/controller/fetchers/schedule_fetcher/schedule_fetcher_new_api.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:uni/controller/fetchers/schedule_fetcher/schedule_fetcher.dart';
+import 'package:uni/controller/networking/network_router.dart';
+import 'package:uni/controller/parsers/schedule/new_api/parser.dart';
+import 'package:uni/model/entities/lecture.dart';
+import 'package:uni/model/entities/profile.dart';
+import 'package:uni/model/entities/session.dart';
+
+/// Class for fetching the user's lectures from the schedule's HTML page.
+class ScheduleFetcherNewApi extends ScheduleFetcher {
+  @override
+  List<String> getEndpoints(Session session) {
+    final urls = NetworkRouter.getBaseUrlsFromSession(session)
+        .map((url) => '${url}hor_geral.estudantes_view')
+        .toList();
+    return urls;
+  }
+
+  /// Fetches the user's lectures from the schedule's HTML page.
+  @override
+  Future<List<Lecture>> getLectures(Session session, Profile profile) async {
+    final endpoints = getEndpoints(session);
+    final lectiveYear = getLectiveYear(DateTime.now());
+
+    final futures = <Future<List<Lecture>>>[];
+
+    for (final baseUrl in endpoints) {
+      final future = Future(() async {
+        final scheduleResponse = await NetworkRouter.getWithCookies(
+          baseUrl,
+          {
+            'pv_num_unico': session.username,
+            'pv_ano_lectivo': lectiveYear.toString(),
+            'pv_periodos': '1',
+          },
+          session,
+        );
+
+        final scheduleApiUrl = getScheduleApiUrlFromHtml(scheduleResponse);
+
+        final scheduleApiResponse = await NetworkRouter.getWithCookies(
+          scheduleApiUrl,
+          {},
+          session,
+        );
+
+        return getLecturesFromApiResponse(scheduleApiResponse);
+      });
+
+      futures.add(future);
+    }
+
+    final results = await Future.wait(futures);
+
+    final lectures = results.expand((element) => element).toList()
+      ..sort((l1, l2) => l1.compare(l2));
```

Comment:
Is this handling repeated classes? Imagine if both faculties have the same class

Reply:
I think that simply doesn't happen in the API. The classes have FEUP in them, which I think means that the class only happens in one faculty. Not sure and we'll have to test that.

Reply:
I think we don't lose anything adding that safety. Otherwise, place a comment, maybe we come back to this in a future ahah